text/0000-dyn-assoc.md

```diff
@@ -80,10 +80,166 @@ fn main() {
 ```
 This can be used by library authors to selectively accept trait objects if certain constants match certain criteria, as well as for introspecting implementation details at runtime, as shown with the floating-point mantissa example.
 
+## Generic contexts
+[generic-contexts]: #generic-contexts
+There's only one case when we don't have direct access to the trait object: compile-time generics. This example demonstrates the issue:
+```rust
+trait Foo {
+    const CONST: usize;
+    
+    fn assoc();
+}
+// Long story short, we're taking a generic argument
+// without a vtable, but we need to do dynamic dispatch.
+fn foo<T: Foo + ?Sized>() {
+    let _ = T::CONST;
+    T::assoc();
+}
+
+fn main() {
+    foo::<dyn Foo>();
+}
+```
+We need to somehow provide `foo<T>()` with a vtable for the trait object, but we don't have one. With the `trait_object.associated_function()` syntax (or the alternatives described in the [unresolved questions][unresolved-questions] section), there's the trait object which we use to perform dynamic dispatch in the calling function — we're calling associated functions and are looking up constants from the **trait** of the trait object, **having the trait object available to also be used to have trait methods called**. But in this case, we don't have a trait object to start with. As a result, if we're calling a function/method/whatever with generic arguments **and the callee uses the associated members** and are passing in `dyn Foo`, the compiler emits an error: "no trait object provided to perform dynamic dispatch", highlighting the exact associated function which requires a trait object with "trait object required here" like some other errors do.
```

Comment:
I see. This indeed looks unsolvable, we might as well just remove the feature with associated members at all, leaving only the `trait_object::impl::associated_member()` syntax, since this one is way too problematic.